### PR TITLE
WIP - MyCoralCrowdsale contract

### PR DESCRIPTION
**Pull request by KristonCosta**
Still WIP - need to make tests and properly configure the deploy.

The crowdsale contract is a timed and whitelisted contract that delivers tokens at the end of the sale.

Installation/Testing:
Install truffle and testrpc:
```
 npm install -g truffle
 npm install -g ethereumjs-testrpc
```
Setting up the repo:
```
git clone git@github.com:mycoralhealth/mycoral-token.git
cd mycoral-token
npm install
```
Deploying the local test network with the first user unlocked
```
testrpc -u 0 
```
Deploying the contract to the network
```
truffle compile
truffle migrate --reset
```

Phases:
Phase 1 - delivering private sale tokens, still need to figure out the best way to handle this.
Phase 2 - We can bulk add whitelisted addresses using MyCoralCrowdSale.addManyToWhitelist([addressList]) and they will be able to buy tokens before the crowdsale "startTime"
Phase 3 - Once the start time passes anyone is able to buy tokens from the crowdsale until the end time

Once the end time is reached people will be able to fetch their tokens using MyCoralCrowdSale.withdrawl()

Test:
Bring up a fresh testrpc and run:
```
truffle test
```

### CODE REVIEW

File: contracts/MyCoralToken.sol [hunk 1]
```diff
@@ -0,0 +1,9 @@
+pragma solidity ^0.4.18;
+
+import "zeppelin-solidity/contracts/token/ERC20/MintableToken.sol";
+
+contract MyCoralToken is MintableToken {
+  string public constant name     = "MyCoralHealthToken";
```
Comment (nosequeldeebee) [inline contracts/MyCoralToken.sol]: sorry, I should have told you. Let's call it `Coral Health Token`

File: contracts/MyCoralToken.sol [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  string public constant symbol   = "MCHT";
```
Comment (nosequeldeebee) [inline contracts/MyCoralToken.sol]: let's make the symbol `CHT`

File: contracts/TimedWhitelistCrowdsale.sol [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import "zeppelin-solidity/contracts/crowdsale/validation/WhitelistedCrowdsale.sol"; 
+import "zeppelin-solidity/contracts/crowdsale/validation/TimedCrowdsale.sol";
+
+contract TimedWhitelistCrowdsale is WhitelistedCrowdsale, TimedCrowdsale {
+	function TimedWhitelistCrowdsale(uint256 _startTime, uint256 _endTime) public
+		TimedCrowdsale(_startTime, _endTime) {
+	}
+
+	function hasStarted() public view returns (bool) {
+		return now > openingTime;
```
Comment (KristonCosta) [inline contracts/TimedWhitelistCrowdsale.sol]: Should be >=

File: contracts/MyCoralCrowdsale.sol [hunk 4]
```diff
@@ -0,0 +1,32 @@
+pragma solidity ^0.4.18;
+
+import "./TimedWhitelistCrowdsale.sol";
+import "zeppelin-solidity/contracts/crowdsale/emission/MintedCrowdsale.sol";
+import "zeppelin-solidity/contracts/crowdsale/distribution/PostDeliveryCrowdsale.sol";
+import "zeppelin-solidity/contracts/token/ERC20/MintableToken.sol";
+
+contract MyCoralCrowdsale is MintedCrowdsale, PostDeliveryCrowdsale, TimedWhitelistCrowdsale {
```
Comment (KristonCosta) [inline contracts/MyCoralCrowdsale.sol]: Need to properly document this contract

File: contracts/MyCoralCrowdsale.sol [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+	modifier onlyWhileNotClosed {
+    	require(now <= closingTime);
+    	_;
+  	}
+
+	// Would be used to transfer tokens that were prepurchased - WIP
+	function addBalanceToAddress(address _beneficiary, uint256 weiAmount)
```
Comment (nosequeldeebee) [inline contracts/MyCoralCrowdsale.sol]: pass in number of Coral Health tokens to send instead of ETH

File: contracts/MyCoralCrowdsale.sol (hunk 4 above)
Comment (nosequeldeebee) [inline contracts/MyCoralCrowdsale.sol]: add max tokens to be distributed

File: contracts/MyCoralToken.sol [hunk 6]
```diff
@@ -0,0 +1,9 @@
+pragma solidity ^0.4.18;
+
+import "zeppelin-solidity/contracts/token/ERC20/MintableToken.sol";
+
+contract MyCoralToken is MintableToken {
+  string public constant name     = "Coral Health Token";
+  string public constant symbol   = "CHT";
+  uint8  public constant decimals = 18;
```
Comment (nosequeldeebee) [inline contracts/MyCoralToken.sol:8]: add field for max tokens

File: contracts/MyCoralToken.sol [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+}
```
Comment (nosequeldeebee) [inline contracts/MyCoralToken.sol:9]: investigate how to deliver tokens upon conclusion of crowdsale. And tokens should be distributed once the cap is hit, even if it is before the end date

**Pull request merged**